src/plugins/alt-titles.ts

```diff
@@ -0,0 +1,217 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  AltTitleNode,
+  ManuscriptEditorView,
+  schema,
+} from '@manuscripts/transform'
+import { Node as ProseMirrorNode } from 'prosemirror-model'
+import { Plugin, PluginKey } from 'prosemirror-state'
+import { Decoration, DecorationSet } from 'prosemirror-view'
+import { arrowDown } from '../icons'
+import { skipTracking } from '@manuscripts/track-changes-plugin'
+
+export interface PluginState {
+  collapsed: boolean
+  titleEnd: number | undefined
+  runningTitle: [AltTitleNode, number] | undefined
+  shortTitle: [AltTitleNode, number] | undefined
+}
+
+function getTitlesData(doc: ProseMirrorNode) {
+  let titleEnd: number | undefined
+  let runningTitle: [AltTitleNode, number] | undefined
+  let shortTitle: [AltTitleNode, number] | undefined
+
+  doc.descendants((node, pos) => {
+    if (titleEnd && runningTitle && shortTitle) {
+      return false
+    }
+    if (node.type === schema.nodes.title && node.nodeSize > 0) {
+      // if title is empty we don't allow to edit alt titles
+      titleEnd = pos + node.nodeSize
+    }
+    if (node.type === schema.nodes.alt_title) {
+      if (node.attrs.type === 'running') {
+        runningTitle = [node as AltTitleNode, pos]
+      }
+      if (node.attrs.type === 'short') {
+        shortTitle = [node as AltTitleNode, pos]
+      }
+    }
+  })
+  return { titleEnd, runningTitle, shortTitle }
+}
+
+function createAltTitlesButton(listener: () => void) {
+  const altTitlesButton = document.createElement('button')
+  altTitlesButton.classList.add('alt-titles-open', 'button-reset')
+  altTitlesButton.innerHTML = arrowDown
+  altTitlesButton.addEventListener('click', (e) => {
+    e.preventDefault()
+    listener()
+  })
+  return altTitlesButton
+}
+
+function selectionInAltTitles(from: number, to: number, state: PluginState) {
+  if (state.runningTitle && state.shortTitle) {
+    const range = {
+      from: Math.min(state.runningTitle[1], state.shortTitle[1]),
+      to: Math.max(
+        state.runningTitle[1] + state.runningTitle[0].nodeSize,
+        state.shortTitle[1] + state.shortTitle[0].nodeSize
+      ),
+    }
+    // if selection overlaps with the range of alt titles, the selection is in the titles
+    return Math.max(from, range.from) <= Math.min(to, range.to)
+  }
+  return false
+}
+
+export const altTitlesKey = new PluginKey<PluginState>('altTitles')
+
+export default () => {
+  return new Plugin<PluginState>({
+    key: altTitlesKey,
+    state: {
+      init: (_, state) => {
+        return { collapsed: true, ...getTitlesData(state.doc) }
+      },
+      apply: (tr, value) => {
+        let newState = value
+        if (tr.docChanged) {
+          newState = { ...newState, ...getTitlesData(tr.doc) }
+        }
+        if (
+          tr.selectionSet &&
+          selectionInAltTitles(tr.selection.from, tr.selection.to, newState)
```

Comment:
Why can't we just depend on `hasParentNodeOfType(schema.nodes.title)(selection)`?

Reply:
I don't mind using this utility here. However my approach uses data that we collect anyway. Also it seems to me the selection has to be completely inside the node for 'hasParentNodeOfType' to return true in this circumstances so it might not be always suitable for selection verification. This is not particularly important in this case but will work even if selection is a bit off.